tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/CombinationListingFeatureContext.php

```diff
@@ -72,88 +73,157 @@ private function fillCommand(UpdateCombinationFromListingCommand $command, array
     }
 
     /**
-     * @Then I should see following combinations of product :productReference in page :page limited to maximum :limit per page:
+     * @Then I should see following combinations list of product ":productReference":
      *
      * @param string $productReference
-     * @param int $page
-     * @param TableNode $dataTable
-     * @param int $limit
+     * @param TableNode $tableNode
      */
-    public function assertCombinationsPage(string $productReference, int $page, TableNode $dataTable, int $limit): void
+    public function assertCombinationsList(string $productReference, TableNode $tableNode): void
     {
-        $offset = $this->countOffset($page, $limit);
-
-        $this->assertPaginatedCombinationList($productReference, $dataTable->getColumnsHash(), $limit, $offset);
+        $this->assertCombinations($productReference, $tableNode->getColumnsHash());
     }
 
     /**
-     * @Then there should be no combinations of :productReference in page :page when limited to maximum :limit per page
+     * @Then combinations list of product ":productReference" should be empty
      *
      * @param string $productReference
-     * @param int $page
-     * @param int $limit
      */
-    public function assertNoCombinationsInPage(string $productReference, int $page, int $limit): void
+    public function assertNoCombinationsInPage(string $productReference): void
     {
-        $offset = $this->countOffset($page, $limit);
+        $this->assertCombinations($productReference, []);
+    }
 
-        $this->assertPaginatedCombinationList($productReference, [], $limit, $offset);
+    /**
+     * @Given product ":productReference" combinations list search criteria is reset to defaults
+     *
+     * @param string $productReference
+     */
+    public function cleanSearchCriteria(string $productReference): void
+    {
+        $this->getSharedStorage()->clear($this->getSearchCriteriaKey($productReference));
     }
 
     /**
-     * Asserts expected product combinations and sets combination references in shared storage
+     * @Transform table:criteria,value
      *
-     * @Then product :productReference should have following list of combinations:
+     * @param TableNode $tableNode
      *
-     * @param string $productReference
-     * @param TableNode $table
+     * @return CombinationFilters
      */
-    public function assertWholeCombinationsList(string $productReference, TableNode $table): void
```

Comment:
I think we should keep this step which is convenient to check the whole list (and it is still used in the generation feature). And create a dedicated one for pagination